artemis/modules/base/base_newer_version_comparer.py

```diff
@@ -0,0 +1,58 @@
+import datetime
+import subprocess
+import tempfile
+from pathlib import Path
+
+import semver
+import yaml
+
+from artemis.module_base import ArtemisBase
+
+
+class BaseNewerVersionComparerModule(ArtemisBase):
+    software_name: str
+
+    def __init__(self, *args, **kwargs):  # type: ignore
+        super().__init__(*args, **kwargs)
+
+        self.endoflife_data_folder = tempfile.mkdtemp()
+        subprocess.call(
+            ["git", "clone", "https://github.com/endoflife-date/endoflife.date", self.endoflife_data_folder]
+        )
+
+        endoflife_data_path = Path(self.endoflife_data_folder) / "products" / (self.software_name + ".md")
+        with open(endoflife_data_path, "r") as f:
+            self.endoflife_data = next(yaml.load_all(f, yaml.SafeLoader))
+
+    def _parse_version(self, version: str) -> semver.VersionInfo:
+        if version.count(".") == 1:
+            return semver.VersionInfo.parse(version + ".0")
+        else:
+            return semver.VersionInfo.parse(version)
+
+    def is_version_obsolete(
+        self,
+        version: str,
+    ) -> bool:
+        version_parsed = self._parse_version(version)
+
+        for release in self.endoflife_data["releases"]:
+            if not (version == release["releaseCycle"] or version.startswith(release["releaseCycle"] + ".")):
```

Comment:
wouldnt `version.startswith(release["releaseCycle"])` be enough?

Reply:
added comment